scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -345,16 +345,12 @@ class FossId internal constructor(
         return ScanSummary(startTime, Instant.now(), "", sortedSetOf(), sortedSetOf(), listOf(issue))
     }
 
-    private suspend fun List<Scan>.findLatestPendingOrFinishedScan(
-        url: String,
-        revision: String? = null
-    ): Scan? =
-        filter {
-            // The scans in the server contain the url with the credentials so we have to remove it for the
-            // comparison. If we don't, the scans won't be matched if the password changes!
-            val urlWithoutCredentials = it.gitRepoUrl?.replaceCredentialsInUri()
-            urlWithoutCredentials == url && (revision == null || (it.gitBranch == revision))
-        }.sortedByDescending { scan -> scan.id }.find { scan ->
+    /**
+     * Find the latest [Scan] in this (already filtered and sorted) list with a finished state. If necessary, wait for
```

Comment:
...(already filtered and sorted)... according which criteria ?
IMO you can remove this comment.

Reply:
Reworded the comment.